x-pack/plugins/security_solution/public/common/components/grouping/container/index.tsx

```diff
@@ -0,0 +1,193 @@
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiSpacer, EuiTablePagination } from '@elastic/eui';
+import type { Filter } from '@kbn/es-query';
+import { isArray } from 'lodash/fp';
+import React, { useMemo, useState } from 'react';
+import { createGroupFilter } from '../accordion_panel/helpers';
+import { tableDefaults } from '../../../store/data_table/defaults';
+import { defaultUnit } from '../../toolbar/unit';
+import type { BadgeMetric, CustomMetric } from '../accordion_panel';
+import { GroupPanel } from '../accordion_panel';
+import { GroupStats } from '../accordion_panel/group_stats';
+import { EmptyGroupingComponent } from '../empty_resuls_panel';
+import { GroupingStyledContainer, GroupsUnitCount } from '../styles';
+import { GROUPS_UNIT } from '../translations';
+import type { GroupingTableAggregation, RawBucket } from '../types';
+
+interface GroupingContainerProps {
+  badgeMetricStats?: (fieldBucket: RawBucket) => BadgeMetric[];
+  customMetricStats?: (fieldBucket: RawBucket) => CustomMetric[];
+  data: GroupingTableAggregation &
+    Record<
+      string,
+      {
+        value?: number | null | undefined;
+        buckets?:
+          | Array<{
+              doc_count?: number | null | undefined;
+            }>
+          | undefined;
+      }
+    >;
+  groupPanelRenderer?: (fieldBucket: RawBucket) => JSX.Element | undefined;
+  groupsSelector?: JSX.Element;
+  inspectButton?: JSX.Element;
+  pagination: {
+    pageIndex: number;
+    pageSize: number;
+    onChangeItemsPerPage: (itemsPerPageNumber: number) => void;
+    onChangePage: (pageNumber: number) => void;
+  };
+  renderChildComponent: (groupFilter: Filter[]) => React.ReactNode;
+  selectedGroup: string;
+  takeActionItems: (groupFilters: Filter[]) => JSX.Element[];
+  unit?: (n: number) => string;
+}
+
+const GroupingContainerComponent = ({
+  badgeMetricStats,
+  customMetricStats,
+  data,
+  groupPanelRenderer,
+  groupsSelector,
+  inspectButton,
+  pagination,
+  renderChildComponent,
+  selectedGroup,
+  takeActionItems,
+  unit = defaultUnit,
+}: GroupingContainerProps) => {
+  const [trigger, setTrigger] = useState<
+    Record<string, { state: 'open' | 'closed' | undefined; selectedBucket: RawBucket }>
+  >({});
+
+  const groupsNumber = data?.groupsNumber?.value ?? 0;
+  const unitCountText = useMemo(() => {
+    const count =
+      data?.alertsCount?.buckets && data?.alertsCount?.buckets.length > 0
+        ? data?.alertsCount?.buckets[0].doc_count ?? 0
+        : 0;
+    return `${count.toLocaleString()} ${unit && unit(count)}`;
+  }, [data?.alertsCount?.buckets, unit]);
+
+  const unitGroupsCountText = useMemo(
+    () => `${groupsNumber.toLocaleString()} ${GROUPS_UNIT(groupsNumber)}`,
+    [groupsNumber]
+  );
+
+  const groupPanels = useMemo(
+    () =>
+      data.stackByMultipleFields0?.buckets?.map((groupBucket) => {
+        const group = isArray(groupBucket.key) ? groupBucket.key[0] : groupBucket.key;
+        const groupKey = `group0-${
```

Comment:
can't this just be `const groupKey = 'group0-${group}'? Same logic as line 87 above